limacharlie/hive.go

```diff
@@ -0,0 +1,268 @@
+package limacharlie
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+type HiveClient struct {
+	Organization *Organization
+}
+
+type HiveArgs struct {
+	HiveName     string
+	PartitionKey string
+	Key          string
+	Data         []byte
+	Expiry       int64
+	Enabled      bool
+	Tags         []string
+	ETag         string
+}
+
+type HiveData struct {
+	Data   map[string]interface{} `json:"data"`
+	SysMtd SysMtd                 `json:"sys_mtd"`
+	UsrMtd UsrMtd                 `json:"usr_mtd"`
+}
+
+type HiveInfo struct {
+	Name      string `json:"name"`
+	Partition string `json:"partition"`
+}
+
+type HiveResp struct {
+	Guid string   `json:"guid"`
+	Hive HiveInfo `json:"hive"`
+	Name string   `json:"name"`
+}
+
+type SysMtd struct {
+	CreatedBy   string `json:"created_by"`
+	Etag        string `json:"etag"`
+	GUID        string `json:"guid"`
+	LastAuthor  string `json:"last_author"`
+	LastError   string `json:"last_error"`
+	LastErrorTs int64  `json:"last_error_ts"`
+	LastMod     int64  `json:"last_mod"`
+}
+type UsrMtd struct {
+	Enabled bool     `json:"enabled"`
+	Expiry  int64    `json:"expiry"`
+	Tags    []string `json:"tags"`
+}
+
+func NewHiveClient(org *Organization) *HiveClient {
+	return &HiveClient{Organization: org}
+}
+
+func (h *HiveClient) List(args HiveArgs, isPrint bool) (map[string]HiveData, error) {
+	var hiveSet map[string]HiveData
+	if err := h.Organization.client.reliableRequest(http.MethodGet,
+		fmt.Sprintf("hive/%s/%s", args.HiveName, args.PartitionKey), makeDefaultRequest(&hiveSet)); err != nil {
+		return nil, err
+	}
+
+	if isPrint {
```

Comment:
I feel like we should leave it to the users to decide how and when they want to print.

Reply:
Cool, I removed the isPrint param from all function calls